modules/personalize_taxonomy_context/personalize_taxonomy_context.module

```diff
@@ -0,0 +1,78 @@
+<?php
+
+/**
+ * @file personalize_taxonomy_context.module
+ * Provides a visitor context plugin for taxonomy.
+ */
+
+/**
+ * Implements hook_personalize_visitor_contexts().
+ */
+function personalize_taxonomy_context_personalize_visitor_context() {
+  $info = array();
+  $path = drupal_get_path('module', 'personalize_taxonomy_context') . '/plugins';
+  $info['taxonomy_context'] = array(
+    'path' => $path . '/visitor_context',
+    'handler' => array(
+      'file' => 'TaxonomyContext.inc',
+      'class' => 'TaxonomyContext',
+    ),
+  );
+  return $info;
+}
+
+/**
+ * Implements hook_node_view().
+ */
+function personalize_taxonomy_context_node_view($node, $view_mode, $langcode) {
+  personalize_taxonomy_context_node_taxonomy($node, $view_mode == 'full');
+}
+
+/**
+ * Handles the node-specific taxonomy data.
+ *
+ * @param $node
+ *   (optional) If passed then the taxonomy terms for the selected node
+ *   are added to the page taxonomy context.
+ * @param bool $primary
+ *   True if this is the primary node displayed for the page, false otherwise.
```

Comment:
Right, it *may* have had a use case in the original function, I don't remember - but don't just blindly copy and paste. It seems to me that personalize_taxonomy_context_node_view() should check whether the view mode is 'full' (that's where this logic belongs) and only call this function if it is.